core/src/main/codegen/templates/Parser.jj

```diff
@@ -1335,7 +1335,7 @@ SqlSelect SqlSelect() :
 }
 {
     <SELECT> { s = span(); }
-    [ <HINT_BEG> AddHint(hints) ( <COMMA> AddHint(hints) )* <COMMENT_END> ]
+    [ <HINT_BEG> (AddHint(hints))? ( <COMMA> AddHint(hints) )* <COMMENT_END> ]
```

Comment:
Calcite also supports `TableHints`, so you will also need to modify it accordingly. WDYT?